### PR TITLE
chore: disable ItemsPresenter Header/Footer on native ListViewBases not only ListViews

### LINKED ISSUE #14543: `GridView` with `Header` and `HeaderTemplate` renders incorrectly starting with 5.0.19
### Current behavior

The header seems to be rendered twice: Once on the left of the grid view (which takes away most of the size), and once on top of the list view where it belongs.

Screenshot_1700772040

### Expected behavior

This is how it looks like up to latest 4.x

Screenshot_1700770660

### How to reproduce it (as minimally and precisely as possible)

Just add a GridView with Header and HeaderTemplate. Watch how header is rendered twice.

### Workaround

-

### Works on UWP/WinUI

Yes

### Environment

_No response_

### NuGet package version(s)

5.0.19 and up

### Affected platforms

Android

### IDE

Visual Studio 2022

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #14543

## PR Type

What kind of change does this PR introduce?
- Bugfix

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 445c4c1</samp>

Fix header and footer layout in `GridView` by updating `HeaderFooterEnabled` property in `ItemsPresenter.cs` to use `ListViewBase` as the templated parent.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsPresenter.cs [hunk 1]
```diff
@@ -31,10 +31,10 @@ namespace Windows.UI.Xaml.Controls
 {
 	public partial class ItemsPresenter : FrameworkElement, IScrollSnapPointsInfo
 	{
-		// TODO: support for Header/Footer when inside a ListView
+		// TODO: support for Header/Footer when inside a ListViewBase
 		private bool HeaderFooterEnabled =>
 #if __ANDROID__ || __IOS__
-		TemplatedParent is not ListView
+		TemplatedParent is not ListViewBase
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsPresenter.cs:37]: Let's add a runtime test that validates that behavior for grid and listview.

**Review by jeromelaban — APPROVED** [review]

**Review by MartinZikmund — APPROVED** [review]

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.0

**Review by MartinZikmund — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsPresenter.cs [hunk 2]
```diff
@@ -31,10 +31,10 @@ namespace Windows.UI.Xaml.Controls
 {
 	public partial class ItemsPresenter : FrameworkElement, IScrollSnapPointsInfo
 	{
-		// TODO: support for Header/Footer when inside a ListView
+		// TODO: support for Header/Footer when inside a ListViewBase
 		private bool HeaderFooterEnabled =>
-#if __ANDROID__ || __IOS__
-		TemplatedParent is not ListView
+#if __ANDROID__
```
[the remark this answers is not in the document]
Comment (ramezgerges) [reply src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsPresenter.cs:36]: I'm not sure what the situation with NativeListViewBase on iOS is. I initially thought it supported headers/footers (like android) but the tests failed indicating that no header was found. So I removed the condition that prevents ItemsPresenter from creating a header/footer from iOS and a different test seems to be failing (the previously failing test now passes). Not sure where to go from here.

[the remark this answers is not in the document]
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Controls/ItemsControl/ItemsPresenter.cs:36]: We need to test the behavior on iOS for sure. If the content is duplicated on iOS, we need to validate the behavior as well.

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ListViewBase.cs [hunk 3]
```diff
@@ -4112,6 +4112,61 @@ public async Task When_Footer_DataContext()
...
+				 		<DataTemplate>
+				 """
+				+
+				"			<TextBlock Text=\"{Binding}\" />"
+				+
+				$"""
+						</DataTemplate>
+					</{listViewBaseType}.HeaderTemplate>
+				</{listViewBaseType}>
+				""");
```
Comment (Youssef1313) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ListViewBase.cs:4141]: Why are we concatenating strings like this? I think it should be possible to have this as a single interpolated raw string literal

Comment (jeromelaban) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ListViewBase.cs:4141]: @ramezgerges @Xiaoy312 does https://github.com/unoplatform/uno/pull/14904 fix this particular PR ?

Comment (Xiaoy312) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_ListViewBase.cs:4141]: 14904 is a replacement of this pr

**Comment by Xiaoy312** [comment]
replaced by #14904

**Pull request closed without merging**